egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -213,6 +238,24 @@ def AddSigmoidLayer(config_lines, name, input, self_repair_scale = None):
     return {'descriptor':  '{0}_sigmoid'.format(name),
             'dimension': input['dimension']}
 
+def AddUnfoldedRnnLayer(config_lines, name, input, output_dim, ng_affine_options = "", self_repair_scale = None):
```

Comment:
The input in your case is not the standard input expected by a method in `components.py`.   I would recommend refactoring this code so that the `AddUnfoldedRnnLayer` works exactly like any other layer, i.e., argument `input` is a normal normal input viz. a  dictionary with a descriptor string and dimension. Make the call to `AddUnfoldedInputLayer` from this method and also rename `AddUnfoldedInputLayer` as this is not a generic method and would just be used in this in unfolded rnns. I would suggest something like `GetInputSplices` which is called for each unfolded step from this method.